sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -729,6 +732,13 @@ public void validate() {
     private Instant curTimestamp;
     private Iterator<PartitionState> curBatch = Collections.emptyIterator();
 
+    private final Counter messagesConsumed = Metrics.counter("io", "messagesConsumed");
+    private final Counter bytesConsumed = Metrics.counter("io", "bytesConsumed");
```

Comment:
I chatted a bit with @aviemzur .

Sounds like a few things should happen:
* defining some standard metrics that are used in generalized contexts. For example, source bytes read or sink elements written. Since many sources would want the same metrics, it would help to standardize them.
* adding Kafka-specific metrics to KafkaIO -- I'd probably use the `KafkaIO.class` or `KafkaIO.{Read/Writer}` as the namespace, and then pick whatever metric names you want.

I'd suggest you summarize this on the list as a proposal. And make a few (very targeted) proposals for specific standardized metric names/namespaces. That will unblock the work in this PR but also get the community involved and lead towards community consensus.

Specifically, don't get too bogged down in e.g. enumerating all possible standard metric namespaces and names. Just the ones for unbounded sources that you want to use here. ;).

What do you think?